plugins/aws_sts/lib/samson_aws_sts/samson_plugin.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module SamsonAwsSts
+  SESSION_DURATION_MIN = 900  # 15 minutes
+  SESSION_DURATION_MAX = 3600 # 1 hour
+
+  class Engine < Rails::Engine
+  end
+
+  def self.sts_client
+    Aws::STS::Client.new(
+      access_key_id:     ENV['SAMSON_STS_AWS_ACCESS_KEY_ID'],
+      secret_access_key: ENV['SAMSON_STS_AWS_ACCESS_SECRET_ACCESS_KEY'],
+      region:            ENV['SAMSON_STS_AWS_REGION'],
+      stub_responses: Rails.env.test?
+    )
+  end
+
+  class Client
+    def initialize(sts_client: SamsonAwsSts.sts_client)
+      @sts_client = sts_client
+    end
+
+    def caller_user_id
+      sts_client.get_caller_identity[:user_id]
+    end
+
+    def deploy_env_vars(deploy:)
+      return {} if deploy.stage.aws_sts_iam_role_arn.blank?
+
+      creds = assume_role(
+        role_arn: deploy.stage.aws_sts_iam_role_arn,
+        role_session_name: session_name(deploy),
+        duration_seconds: deploy.stage.aws_sts_iam_role_session_duration
+      ).credentials
+
+      {
+        STS_AWS_ACCESS_KEY_ID:     "hidden://#{creds.access_key_id}",
+        STS_AWS_SECRET_ACCESS_KEY: "hidden://#{creds.secret_access_key}",
+        STS_AWS_SESSION_TOKEN:     "hidden://#{creds.session_token}"
+      }
+    end
+
+    def assume_role(role_arn:, role_session_name:, duration_seconds: SESSION_DURATION_MIN)
+      sts_client.assume_role(
+        role_arn: role_arn,
+        role_session_name: role_session_name,
+        duration_seconds: duration_seconds
+      )
+    end
+
+    private
+
+    attr_reader :sts_client
+
+    def session_name(deploy)
+      project_name = deploy.project.name.parameterize[0..15]
+      stage_name   = deploy.stage.name.parameterize[0..15]
+
+      "#{project_name}:#{stage_name}:deploy/#{deploy.id}"
+    end
+  end
+end
+
+Samson::Hooks.view :stage_form, 'samson_aws_sts/fields'
+
+Samson::Hooks.callback :stage_permitted_params do
+  %i[
+    aws_sts_iam_role_arn
+    aws_sts_iam_role_session_duration
+  ]
+end
+
+Samson::Hooks.callback :deploy_env do |deploy|
+  SamsonAwsSts::Client.new(sts_client: SamsonAwsSts.sts_client).deploy_env_vars(deploy: deploy)
```

Comment:
same here, will break every deploy when not configured correctly ?

Reply:
also should only do any work if the arn is set on the stage so users can opt in/out easily

Reply:
Added a guard clause with `next`, return was causing a localjumperror from the `fire` method in the tests